assets/js/base/components/payment-methods/payment-methods.js

```diff
@@ -38,20 +44,48 @@ const createTabs = ( paymentMethods ) => {
 		: [ noPaymentMethodTab() ];
 };
 
+/**
+ * Returns a payment method for the given context.
+ *
+ * @param {string} id The payment method slug to return.
+ * @param {Object} paymentMethods The current registered payment methods
+ * @param {boolean} isEditor Whether in the editor context (true) or not (false).
+ *
+ * @return The payment method matching the id for the given context.
+ */
+const getPaymentMethod = ( id, paymentMethods, isEditor ) => {
+	let paymentMethod = paymentMethods[ id ] || null;
+	if ( paymentMethod ) {
+		paymentMethod = isEditor
+			? paymentMethod.edit
+			: paymentMethod.activeContent;
+	}
+	return paymentMethod;
+};
+
 const PaymentMethods = () => {
 	const [ checkoutData ] = useCheckoutData();
+	const { isEditor } = useCheckoutContext();
 	const { dispatch, select } = usePaymentEvents();
 	const { isInitialized, paymentMethods } = usePaymentMethods();
+	const currentPaymentMethods = useRef( paymentMethods );
+
+	// update ref on changes
+	useEffect( () => {
+		currentPaymentMethods.current = paymentMethods;
+	}, [ paymentMethods ] );
+
 	const {
 		activePaymentMethod,
 		setActivePaymentMethod,
 	} = useActivePaymentMethod();
 	const getRenderedTab = useCallback(
 		() => ( selectedTab ) => {
-			const paymentMethod =
-				( paymentMethods[ selectedTab ] &&
-					paymentMethods[ selectedTab ].activeContent ) ||
-				null;
+			const paymentMethod = getPaymentMethod(
+				currentPaymentMethods.current,
+				selectedTab,
+				isEditor
+			);
```

Comment:
I think the order of the arguments is not correct here. Shouldn't `selectedTab` be the first one?

Reply:
ooo good catch!